lib/drives/pages/drive_detail_page.dart

```diff
@@ -353,14 +385,39 @@ class _DriveDetailPageState extends State<DriveDetailPage> {
     await _drive?.cancel();
     setState(() {});
   }
+
+  int _getNewMessagesCount(Drive drive) {
+    List<Ride> approvedRides = drive.approvedRides!;
```

Comment:
this could be written more succinctly (basically as a one-liner) using `reduce`: https://api.flutter.dev/flutter/dart-collection/ListQueue/reduce.html

Reply:
also please make this a method on the drive itself

Reply:
thanks for the hint that's way better